packages/docs/pages/StatefulTable/StatefulTablePage.tsx

```diff
@@ -223,6 +229,32 @@ const StatefulTablePage = () => {
         />
         {/* jsx-to-string:end */}
       </CodePreview>
+
+      <H2>Usage with search</H2>
+
+      <CodePreview scope={{ items }}>
+        {/* jsx-to-string:start */}
+        {function Example() {
+          const filter = (value, items) => items.filter((item) => item.name.includes(value));
+
+          return (
+            <StatefulTable
+              itemName="Products"
+              columns={[
+                { header: 'Sku', hash: 'sku', render: ({ sku }) => sku },
+                { header: 'Name', hash: 'name', render: ({ name }) => name },
+                { header: 'Stock', hash: 'stock', render: ({ stock }) => stock, sortKey: 'stock' },
+              ]}
+              items={items}
+              pagination
+              selectable
+              stickyHeader
+              search={{ filter }}
```

Comment:
Not sure if I'm a fan of this API. For `filters` and how you do it on the `Table` component it makes sense to pass in an object, but for the case of the `StatefulTable` maybe we can simplify it further by just passing in a function? Thoughts?

Reply:
I discussed it with @chanceaclark and came to the [conclusion](https://github.com/bigcommerce/big-design/pull/521#discussion_r617627764)